agent/src/main/java/com/teamscale/jacoco/agent/Agent.java

```diff
@@ -98,7 +101,18 @@ private void reuploadCoverageFromPropertiesFile(File file, IUploader uploader) {
 			properties.load(reader);
 			CoverageFile coverageFile = new CoverageFile(
 					new File(StringUtils.stripSuffix(file.getAbsolutePath(), RETRY_UPLOAD_FILE_SUFFIX)));
-			uploader.reupload(coverageFile, properties);
+
+			if (uploader instanceof TeamscaleUploader) {
+				((TeamscaleUploader) uploader).reupload(coverageFile, properties);
```

Comment:
i actually meant to have a e.g. `IUploadRetry` which you can implement in `TeamscaleUploader`, `ArtifactoryUploader` and `AzureFileStorageUploader` and check for here. Otherwise you'd have to remember to also adjust this code when adding a e.g. S3Uploader, which is rather prone to being overlooked.
```
			if (uploader instanceof IUploadRetry) {
				((IUploadRetry) uploader).reupload(coverageFile, properties);
```